### PR TITLE
Add module limits explainer

### PR DESCRIPTION
**Pull request by jlwllmr**
Adds a new page to the prover architecture section to explain module limits. A necessary addition to provide context for methods such as `eth_sendRawTransaction` and `get_transactionExclusionStatusV1`.

### CODE REVIEW

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Regardless of its purpose, each module has a specific limit to the amount of lines of data that 
+its operations can generate. You can view each module's limits in the [Linea source code](https://github.com/Consensys/linea-monorepo/blob/main/config/common/traces-limits-besu-v2.toml).
+Every time one of the operations in a module is used, it generates additional lines.
+
+Transactions with a high complexity—using a large volume of operations repeatedly—may generate too 
+many lines of data, and exceed that module's limit. Transactions that reach this point are 
+rejected by the [sequencer](../sequencer/index.mdx) to ensure that the trace data passed to the 
+prover is of a manageable scope, and that a proof can be generated.
+
+Transaction rejections caused by module limits are prevented at one of three stages:
```
Comment (Ludcour) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:25]: Three stages might be a bit missleading. I'd say instead: "can be identified with three different methods"

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+1. [`linea_estimateGas`](../../../developers/reference/api/linea-estimategas.mdx), which simulates
+a transaction using given parameters. Transactions that exceed module limits cause an error.
+2. [`eth_sendRawTransaction`](../../../developers/reference/api/eth-sendrawtransaction.mdx). 
+Similarly, `eth_sendRawTransaction` simulates the transaction before it is submitted, and 
+returns an error if module limits would be exceeded.
+3. As a final protection, the [`linea_getTransactionExclusionStatusV1`](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx)
```
Comment (Ludcour) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:31]: As a final protection > Finally, if the transaction was sent and doesn't get rejected directly nor included, the [...]
(to be validated by Julien)

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+This mechanism involves module limits. 
+
+Modules are the sections into which Linea's arithmetization specification is divided. Each module 
+represents a collection of operations (opcodes) at the EVM level, or corresponds to a specific 
+function of Linea's zkEVM. Some of the modules process transaction data; some manage zkEVM memory; 
+others coordinate interactions between modules.
+
+Regardless of its purpose, each module has a specific limit to the amount of lines of data that 
+its operations can generate. You can view each module's limits in the [Linea source code](https://github.com/Consensys/linea-monorepo/blob/main/config/common/traces-limits-besu-v2.toml).
+Every time one of the operations in a module is used, it generates additional lines.
```
Comment (OlivierBBB) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:18]: It _may_ generate additional lines, this depends on the module. For instance for modules that deal with 'simple' instructions à la `ADD`, `SUB`, `MUL`, `EXP`, `AND`, `OR`, `XOR`, `NOT`, `BYTE`, `SIGNEXTEND`, `SHL`, `SHR`, `SAR`, `MOD`, `DIV`, `SMOD`, `SDIV`, `ADDMOD`, `MULMOD`, ... among others only get traced once so repeat calls with the same arguments don't get traced multiple times.

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx (hunk 2 above)
Comment (jlwllmr) [reply docs/architecture/stack/trace-expansion-proving/prover-limits.mdx]: @julien-marchand could you check the suggestion please?

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+a transaction using given parameters. Transactions that exceed module limits cause an error.
+2. [`eth_sendRawTransaction`](../../../developers/reference/api/eth-sendrawtransaction.mdx). 
+Similarly, `eth_sendRawTransaction` simulates the transaction before it is submitted, and 
+returns an error if module limits would be exceeded.
+3. Finally, if the transaction was sent and doesn't get rejected directly, nor included in a block, 
+the [`linea_getTransactionExclusionStatusV1`](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx)
+API method can be used to query the reason for transaction rejection in edge cases where 
+`linea_estimateGas` or `eth_sendRawTransaction` did not return an error. `linea_getTransactionExclusionStatusV1`
+is only available for Infura nodes via request. View its [reference page](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx) 
+for more information on requesting it.
```
Comment (bgravenorst) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:37]:
```suggestion
Transactions rejected for exceeding module limits can be identified through the following
methods, which return an error:

- [`linea_estimateGas`](../../../developers/reference/api/linea-estimategas.mdx)
- [`eth_sendRawTransaction`](../../../developers/reference/api/eth-sendrawtransaction.mdx)

Use [`linea_getTransactionExclusionStatusV1`](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx)
if a transaction was sent and didn't get rejected directly, nor included in a block.
This API method can be used to query the reason for transaction rejection in edge cases where
`linea_estimateGas` or `eth_sendRawTransaction` did not return an error.

:::note
The [`linea_getTransactionExclusionStatusV1`](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx)
API method is only available with Infura, and you need to [request access to the method](https://support.infura.io/).
:::
```
There is no mention of requesting access to the method in the existing content, so I linked to Infura support.

File: docs/developers/reference/api/eth-sendrawtransaction.mdx [hunk 5]
```diff
@@ -31,7 +31,7 @@ On Linea, each transaction is represented by a sequence of lines representing th
 happening at the EVM level. These lines are grouped by modules and are used as inputs to the prover. 
 In order to limit the resources required to generate proofs, limits are enforced on the maximum 
 number of lines each module can have. Hence, when a transaction is considered for inclusion in a 
-block, a line count check is perform to ensure it conforms to the requirements.
+block, a line count check is perform to ensure it conforms to the requirements. [Read more](../../../architecture/stack/trace-expansion-proving/prover-limits.mdx).
```
Comment (bgravenorst) [inline docs/developers/reference/api/eth-sendrawtransaction.mdx:34]:
```suggestion
block, a [line count check is performed to ensure it conforms to the requirements](../../../architecture/stack/trace-expansion-proving/prover-limits.mdx).
```

**Review by bgravenorst — APPROVED** [review]
Some feedback

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+manageable scope, and that a proof can be generated.
+
+Transactions rejected for exceeding module limits can be identified through the following
+methods, which return an error:
+
+- [`linea_estimateGas`](../../../developers/reference/api/linea-estimategas.mdx)
+- [`eth_sendRawTransaction`](../../../developers/reference/api/eth-sendrawtransaction.mdx)
+
+Use [`linea_getTransactionExclusionStatusV1`](../../../developers/reference/api/linea-gettransactionexclusionstatusv1.mdx)
+if a transaction was sent and didn't get rejected directly, nor included in a block.
```
Comment (julien-marchand) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:34]:
```suggestion
to get the rejection status of a transaction hash, if it was sent and didn't get rejected directly, nor included in a block.
```

File: docs/architecture/stack/trace-expansion-proving/prover-limits.mdx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+This API method can be used to query the reason for transaction rejection in edge cases where
+`linea_estimateGas` or `eth_sendRawTransaction` did not return an error.
```
Comment (julien-marchand) [inline docs/architecture/stack/trace-expansion-proving/prover-limits.mdx:36]:
```suggestion
`linea_estimateGas` or `eth_sendRawTransaction` did not return an error. It's expected to be used only in cases the transaction execution differs between an RPC node and the sequencer.
```

**Pull request merged**